test/groovy/com/sap/piper/ConfigurationMergerTest.groovy

```diff
@@ -63,6 +63,17 @@ class ConfigurationMergerTest {
         Assert.assertEquals(null, merged.veggie)
     }
 
+    @Test
+    void testMergeDeepStructureWithMissingDefaults(){
+        Map defaults = [others:[apples: 18]]
+        Map configuration = [fruits: [bananaaas: 50, cucumbers: 1000]]
+        Set configurationKeys = ['fruits']
```

Comment:
Does this setting influence the test? I assume, this could be a random string and the assertions were still true, right?

Reply:
No. It has to be exactly the key of the configuration which should be merged in. Otherwise merged.fruits.cucumbers will not exist.